packages/wp-now/src/run-cli.ts

```diff
@@ -31,6 +31,16 @@ function commonParameters(yargs) {
 		.option('php', {
 			describe: 'PHP version to use.',
 			type: 'string',
+		})
+		.option('siteurl', {
+			describe:
+				"Custom site URL to access the site ('home' and 'siteurl' option values).",
+			type: 'string',
+		})
+		.option('customport', {
+			describe:
+				"Custom port number. Needed if you're using a tunneling service (e.g. ngrok), usually 80.",
+			type: 'number',
```

Comment:
I wonder if we can/should remove `customport` and consider it always part of `siteurl`.
Assuming that if a given `siteurl` is passed the custom port will be `80` and if it's a different port, it needs to be explicit on `siteurl` string.

Reply:
I'd prefer this too.

Reply:
After thinking about it, I guess we could take that port number from the URL, and get rid of the `customport` option.
That would mean 3 possible states:
- port 80: no port in the final URL for tunneling service
- non-80 port specified: the final URL will contain that port for tunneling service with non-80 port for some reason
- no port: port will be automatically added to the URL by wp-now

The whole situation is a bit user-unfriendly though, both with `customport` option and without it.

May be we should replace it with a flag (e.g. `--proxying`) to indicate that the final URL does not need to get the port added to it?
That wouldn't work for proxying via non-80 ports, but I don't even know if that's a thing. That potential flexibility may not be worth the added complexity.

What do you think about that?

Reply:
The more that I think about this, the more I feel we should solve this with Blueprints (https://github.com/WordPress/playground-tools/issues/9).

@adamziel Could you provide a sample Blueprint that would achieve what @sergeymitr wants?

Reply:
Created a GitHub issue for @sejas to work on the Blueprint integration: https://github.com/WordPress/playground-tools/issues/80

Reply:
Is this just to define the WordPress constants? If so, this should work:

```json
{
    "steps": [
         { "step": "defineSiteUrlStep", "siteUrl": "http://myurl.localhost" }
     ]
}
```

If you need some extra logic there you could either have three files or generate that blueprint in a custom startup script.

Reply:
Thanks @adamziel.

After additional research, I don't think blueprints would solve the issue entirely:
- port number is being determined dynamically later in the code, so we'd still need to edit the `getWpNowConfig()` function to add it to the URL
- decision whether to add it to the URL (for local domains) or not (for reverse proxies/tunneling services)

Considering changes in the code will still be needed, blueprints don't bring significant improvement. 
Since you're striving toward keeping the tools lean (I fully support this), I suggest one of the approaches below.
I believe they would bring the needed flexibility to solve a lot of edge cases that users will come up with:

1. Do not define the constants `WP_HOME` and `WP_SITEURL`, and save them in the database options table just like regular WordPress installations do.
2. Depending on a flag (e.g. `--extract-url-config`) or some blueprint configuration, add `WP_HOME` and `WP_SITEURL` into the existing `wp-config.php`, or create a new one.
3. Use a different flag (e.g. `--skip-url-config`) or a blueprint to skip setting `WP_HOME` and `WP_SITEURL`. Thus, the site will not work until they are defined in a custom `wp-config.php`.
4. Use another flag (e.g. `--auto-url-config`) or a blueprint to make `WP_HOME` and `WP_SITEURL` take whatever the value is in the environment variables ([example from Jetpack dev env](https://github.com/Automattic/jetpack/blob/ecb8f7d24fa4dde0831f84b0f4d7b6af9a086c72/tools/docker/bin/run.sh#L51-L56)).

It's likely that the issue of hardcoded `WP_HOME` and `WP_SITEURL` will eventually come up, as not being able to change the URL setting of the site limits the ability to develop and test quite a bit. Right now the `wp_option` records get ignored, trying to customize it in `wp-config.php` throws the "Constant already defined" error.

What do you think if we implement one of those approaches, or may be you have other ideas?

I believe being able to edit those values via `wp-config.php` as intended by WordPress would be a better approach than having to implement and apply blueprints to do the same.

Reply:
Hi @sejas.

Speaking of my task at hand, the blueprints would allow using tunneling services, so we're good here.
However, I don't think they'll solve the issue entirely.

WP-now starts a server on a random port (even if you provide port number, it may change if the port is closed). Therefore, custom local URL's will look like this: `http://something.local:8765`. Since we cannot know the port number before starting the server, we'll still need to edit the code to append the port number to the URL provided in the blueprint.

However, tunneling services (ngrok, JT) proxy requests straight to that custom port, making you access the site via port 80. So, the website URL will look like `http://something-ngrok.tld`, with no port number needed.

Blueprints should work for the tunneling services, since there's no need to add port number to the site URL. However, they will not work for regular custom domains because we need to know the port in advance, or append it to the URL. If we go with the latter, we also need to somehow inform the script whether we actually need to append the port.

It also limits the users because of the non-native way to define the site URL. One thing that comes to mind is running multi-URL sites. While `wp-config.php` allows us to simply use environment variables to make the site work on any domain, blueprints don't do that.
They'll also need to restart the server every time the domain changes.

Blueprints are undoubtedly a great way to customize the server, but I still think making `WP_HOME` and `WP_SITEURL` immutable is quite limiting.

Reply:
- @sergeymitr I merged https://github.com/WordPress/playground-tools/pull/82 , It's working on Node v20. Feel free to test it. We didn't release the package yet.

> WP-now starts a server on a random port (even if you provide port number, it may change if the port is closed). Therefore, custom local URL's will look like this: http://something.local:8765. Since we cannot know the port number before starting the server, we'll still need to edit the code to append the port number to the URL provided in the blueprint.

The default port is not random, it's usually `8881`. You are right that this port could change if that port is already in use, but it increases sequentially. So it's a deterministic way and users can predict what port will run wp-now.

Since the port is predictable, the user can write the given port manually and include it on WP_SITEURL through a blueprint.
I'm still thinking that passing the `--url` parameter is something that can be handy for wp-now, but this `url` can also include the default port or the same port passed to `--port` e.g. `--url=http://myurl.wp-now:8881` or `--url=http://myurl.wp-now --port=80`

> Blueprints are undoubtedly a great way to customize the server, but I still think making WP_HOME and WP_SITEURL immutable is quite limiting.

I won't be available next week, but we can keep iterating this PR and probably accept `--url` for a fast option change.

Reply:
Hi @sejas.
Your blueprint PR works great, thanks for implementing that!
It would come handy if we could edit `WP_HOME` and `WP_SITEURL` right in `wp-config.php`, but that's no longer a priority since everything works with blueprints.
I'm going to close the PR now as the changes no longer necessary, and we can revisit this later if needed.